### PR TITLE
Change some Formal Names in Path Module

### PR DESCRIPTION
**Pull request by Maxrimus**
The goal of this PR is to make the argument names in the Path module consistent. I have done the following:
-Changed any arguments named 'name' to 'path'
-Deprecated old procs using 'name' argument
-Changed documentation to reflect new argument names
-Created deprecation tests to confirm all deprecated methods are behaving properly

The code is finished and the Path tests are succeeding, I still need to run the paratest before finalizing this PR.

### CODE REVIEW

**Comment by bradcray** [comment]
@Maxrimus : From the failed check here, it looks like you missed the section about DCOs in the ContributorInfo document.  See step 4 here:  https://github.com/chapel-lang/chapel/blob/master/doc/rst/developer/bestPractices/ContributorInfo.rst#get-set-up

File: modules/standard/Path.chpl [hunk 1]
```diff
@@ -120,20 +120,27 @@ private inline proc unescape(str: string) {
...
-  return normPath(name);
+  if !isAbsPath(path) then
+    return normPath(joinPath(try here.cwd(), path));
+  return normPath(path);
+}
+
+pragma "no doc"
+pragma "last resort"
+proc absPath(name: string): string throws {
+  compilerWarning('Argument `name` is deprecated - use `path` instead');
```
Comment (e-kayrakli) [inline modules/standard/Path.chpl:142]: Can you prefix the deprecation message with the function name? Maybe even with `Path.` qualifier? So it'd be something like:

```
Path.absPath: Argument `name` is deprecated - use `path` instead
```

(Please do the same for others)

Comment (e-kayrakli) [reply modules/standard/Path.chpl:142]: In case this was something you were wondering: in Chapel you can have either single- or double- quoted strings. There's no major difference between the two, and you can use the other-kind-of-quote without escaping in them. So, you could also do this by either:

```chapel
compilerWarning('Argument "name" is deprecated - use "path" instead');
compilerWarning("Argument 'name' is deprecated - use 'path' instead");
```

Not to suggest you make any such change now :)

File: modules/standard/Path.chpl [hunk 2]
```diff
@@ -592,36 +627,51 @@ proc normPath(name: string): string {
...
-proc realPath(out error: syserr, name: string): string {
+pragma "last resort"
+proc realPath(name: string): string throws {
+  compilerWarning('Argument `name` is deprecated - use `path` instead');
+  return realPath(path=name);
+}
+
+
+pragma "no doc"
+proc realPath(out error: syserr, path: string): string {
```
Comment (e-kayrakli) [inline modules/standard/Path.chpl:670]: I realize this is not part of this PR but I would just drop these `out error` versions altogether. I believe they are very old.

Note that there might be some tests that are still using them. I'd run paratest *after* removing these.

File: modules/standard/Path.chpl [hunk 3]
```diff
@@ -701,35 +758,35 @@ proc commonPrefixLength(const a1: [] string, const a2: [] string): int {
...
   :throws SystemError: Upon failure to get the current working directory.
 */
-proc relPath(name: string, start:string=curDir): string throws {
+proc relPath(path: string, start:string=curDir): string throws {
   const realstart = if start == "" then curDir else start;
 
   // NOTE: Reliance on locale.cwd() can't be avoided.
   const startComps = absPath(realstart).split(pathSep, -1, true);
-  const nameComps = absPath(name).split(pathSep, -1, true);
+  const pathComps = absPath(path).split(pathSep, -1, true);
```
Comment (e-kayrakli) [inline modules/standard/Path.chpl:778]: thanks for `nameComps`->`pathComps` change :)

File: modules/standard/Path.chpl (hunk 2 above)
Comment (e-kayrakli) [reply modules/standard/Path.chpl:670]: If you have already run some testing, you can leave them in for now.

File: test/deprecated/formalNameChange.chpl [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+var path = "/";
+var error:syserr;
+
+absPath(name=path);
+basename(name=path);
+dirname(name=path);
+isAbsPath(name=path);
+normPath(name=path);
+realPath(name=path);
+realPath(error,name=path);
```
Comment (e-kayrakli) [inline test/deprecated/formalNameChange.chpl:13]: don't forget to drop this, if you remove the pertinent overload

**Review by e-kayrakli — COMMENTED** [review]
Looking good! I got some minor comments inline

**Comment by Maxrimus** [comment]
@e-kayrakli I've made all the changes you requested, let me know if there's anything else you think needs to be changed. I'm going to run the paratest now.

**Comment by Maxrimus** [comment]
Screen Shot 2021-01-28 at 12 37 57 PM
1 New Passing Deprecation Test
23 Existing Passing Path Tests
12349 Existing Passing Paratest Tests

File: modules/standard/Path.chpl [hunk 5]
```diff
@@ -527,16 +555,23 @@ private proc joinPath(paths: [] string): string {
...
   return result;
 }
 
+pragma "no doc"
+pragma "last resort"
+private proc normalizeLeadingSlashCount(name: string): int {
+  compilerWarning("Path.normalizeLeadingSlashCount: Argument 'name' is deprecated - use 'path' instead");
+  return normalizeLeadingSlashCount(path=name);
+}
+
```
Comment (e-kayrakli) [inline modules/standard/Path.chpl:574]: No need to change anything here, but I am only realizing now that this was actually a private function. These are not user-facing, and we can change them more freely, and without doing real deprecation at all. No harm to have the deprecated version, either, though

**Review by e-kayrakli — APPROVED** [review]
